### PR TITLE
[GH-606] Make pools damage the crates

### PR DESCRIPTION
**Pull request by Nico-Sanchez**
## Motivation

Pools can't damage the crates at this moment. We should fix it.
Closes #610

## Summary of changes

- Wrap entities behavior in Entities module to delegate the action to each category module (player, crate, etc).
- Add crates entities to remove_owner_effects/3 function.
- [Refactor effects mechanics to trigger on entities and leave the behavior to each category](https://github.com/lambdaclass/mirra_backend/pull/610/commits/2bd20dc1c9a5995a9a6115506530f8583a9e9e67)

## How to test it?

Just play a game with the front branch -> `add-crate-feature`

## Checklist
- [x] Tested the changes locally.
- [x] Reviewed the changes on GitHub, line by line.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: apps/arena/lib/arena/game/effect.ex [hunk 1]
```diff
@@ -177,67 +187,68 @@ defmodule Arena.Game.Effect do
...
 
-      %{position: pool_position} when pool_position == player.position ->
-        player
+      %{position: pool_position} when pool_position == entity.position ->
+        entity
 
       pool ->
-        if player.aditional_info.damage_immunity do
-          player
+        if entity.aditional_info.damage_immunity do
```
Comment (tvillegas98) [inline apps/arena/lib/arena/game/effect.ex:207]:
```suggestion
        if Map. has_key?(entity.aditional_info, :damage_immunity) do
```

If the entity doesn't have that key, the game crashes

**Review by tvillegas98 — APPROVED** [review]
The refactor looks good!

**Review by agustinesco — CHANGES_REQUESTED** [review]
Nice refactor!

I'm getting an error when i ult the crates with valtimer

```
** (KeyError) key :collides_with not found in: %{
lib/arena/game/effect.ex:223: Arena.Game.Effect.do_effect_mechanics/4
```

File: apps/arena/lib/arena/game/effect.ex [hunk 2]
```diff
@@ -140,35 +152,33 @@ defmodule Arena.Game.Effect do
...
-        end)
+    Enum.reduce(entities_to_apply, game_state, fn {_entity_id, entity}, game_state ->
+      if Entities.alive?(entity) do
+        entity =
+          Enum.reduce(entity.aditional_info.effects, entity, fn effect, entity ->
+            apply_effect_mechanic(entity, effect, game_state)
+          end)
 
-      put_in(game_state, [:pools, pool.id], pool)
+        Entities.update_entity(entity, game_state)
```
[the remark this answers is not in the document]
Comment (Nico-Sanchez) [reply apps/arena/lib/arena/game/effect.ex:171]: You're right. Done!

**Review by agustinesco — CHANGES_REQUESTED** [review]
Valtimer ultimate isn't pulling people anymore

**Comment by Nico-Sanchez** [comment]
> Valtimer ultimate isn't pulling people anymore

Fixed!

**Review by agustinesco — DISMISSED** [review]
LGTM!

**Comment by manucamejo** [comment]
@Nico-Sanchez please create the front-end PR

**Review by manucamejo — APPROVED** [review]

**Review by manucamejo — CHANGES_REQUESTED** [review]
If we are not using the effect in the front, we shouldn't add it to the messages.proto, please remove it.

**Review by agustinesco — APPROVED** [review]

**Review by tvillegas98 — APPROVED** [review]

**Pull request merged**